app/services/sign_in/state_payload_jwt_encoder.rb

```diff
@@ -29,6 +31,10 @@ def validate_pkce_params
       validate_code_challenge
     end
 
+    def validate_scope
```

Comment:
🤔 I wonder if it would be worthwhile cleaning up all of the `validate_*` methods to include bangs in the method definition to denote the method can raise an exception.

So, `validate_scope` would become `validate_scope!`, and so on. Definitely outside of the scope of this PR, but might be worth considering to be more idiomatic.

Reply:
I went ahead and added `!` to these methods, generally I don't care much about private methods but any possibility for clarity is good, especially if it's this easy to add